### DISCUSSION TITLE
用的1.3B CPU版本llama.cp，为什么推理出来的结果都是无逻辑性的中文

### DISCUSSION THREAD

**Discussion by astarxixi** [ymcui/Chinese-LLaMA-Alpaca-2#437] [Q&A]
用的1.3B CPU版本llama.cp，为什么推理出来的结果都是无逻辑性的中文
![Uploading DB182CFD-CF50-4859-83FC-EF11BA0FCCB8.png…]()

**Comment by ymcui** [comment]
看不到你的图片。另外，你用的是哪个模型？

**Reply by astarxixi** [reply]
用的ggml-model-q4_0.gguf ，执行 sh chat.sh ../chinese-alpaca-2-1.3b-hf/ggml-model-q4_0.gguf '中国首都在哪里'
输出结果：
llama_new_context_with_model: n_ctx      = 4096
llama_new_context_with_model: freq_base  = 10000.0
llama_new_context_with_model: freq_scale = 1
llama_new_context_with_model: kv self size  =  256.00 MiB
llama_build_graph: non-view tensors processed: 96/96
ggml_metal_init: allocating
ggml_metal_init: found device: Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: picking default device: Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: default.metallib not found, loading from source
ggml_metal_init: loading '/Users/admin/Desktop/bairong/AILAB/dev/LLM/llama.cpp/ggml-metal.metal'
ggml_metal_init: GPU name:   Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: hasUnifiedMemory              = true
ggml_metal_init: recommendedMaxWorkingSetSize  =  1536.00 MiB
ggml_metal_init: maxTransferRate               = built-in GPU
llama_new_context_with_model: compute buffer total size = 291.07 MiB
llama_new_context_with_model: max tensor size =   177.19 MiB
ggml_metal_add_buffer: allocated 'data            ' buffer, size =   734.26 MiB, (  738.68 /  1536.00)
ggml_metal_add_buffer: allocated 'kv              ' buffer, size =   256.00 MiB, (  994.68 /  1536.00)
ggml_metal_add_buffer: allocated 'alloc           ' buffer, size =   288.01 MiB, ( 1282.69 /  1536.00)

system_info: n_threads = 8 / 4 | AVX = 1 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | 
main: interactive mode on.
Input prefix with BOS
Input prefix: ' [INST] '
Input suffix: ' [/INST]'
sampling: 
	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	top_k = 40, tfs_z = 1.000, top_p = 0.900, min_p = 0.050, typical_p = 1.000, temp = 0.500
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
generate: n_ctx = 4096, n_batch = 512, n_predict = -1, n_keep = 0


== Running in interactive mode. ==
 - Press Ctrl+C to interject at any time.
 - Press Return to return control to LLaMa.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.

 [INST] <<SYS>>
You are a helpful assistant. 你是一个乐于助人的助手。
<</SYS>>

中国首都在哪里 [/INST]身姿态形态同身双与身双实心理托义柴与与跟跟#同团队门槛蛊跟你基础文化与跟辛与角色门槛代跟与公材料公门槛代与跟文化团队纯产品质量与其归担讨与透明公框跟我讨臂公作为一个书法主题团队团队组织传统文化跟与书籍门槛门槛是我文化公代стори是我国家跟甘参基础设施到了自主跟我公与向代讨我跟时代的门槛框讨跟你与代兼跟跟我跟你代与代跟代火公OT跟我控制代产品团队普遍组织古代材料跟分化框兼具步伐跟产品质量门槛经典控制示范硬件适合团队主题框奈产品古代代控制代是我传统文化框跟框透明стори透明我可以OT古代组织古代基础设施肩膀我跟跟组织到了团队跟国家团队代兼产品质量团队团队代跟你团队微框门槛框OT集体代门槛透明文化产品的兼我跟跟产品讨面向奈团队OT代古代比例示范纯代跟我我跟主题团队臂自主团队甘OT奈框代兼团队控制古代兼公团队便利我跟适合OT是我OT倍硬件我跟跟%,自治匹配装备团队乎并购控制跟你肩膀示范组织透明控制笑替代基础设施观光集体团队为主明显框品种代微门槛框兼团队舒适我跟стори友好经典我跟古代公担是我臂团队示范主题产品质量管控得很控制团队兼具姿势团队肩自主我跟兼框框OT控制主题团队分化团队门槛适合透明渠道%,自治信息亲切步伐团队自主匹配乎畔自治兼控制代透明控制并购团队古代示范古代为主比例装备兼具我都电动硬件兼舒适两个字透明姿态门槛门槛兼到了产品的组织框品种门槛到了hren肩膀檔跟你古代姿势缀】政务经典到了%,门槛产品质量器械我可以自主控制得很管控匹配棚产品便利我跟乎自主示范赏控制团队OThren兼顾兼观光奈代为主臂突集体兼信息并购门口适合装备充实透明度过示范契廉氛围自主团队示范品种水面主题主流比例担自治匹配hren事故笑透明框示范是我стори】步伐透明了解一下示范监护如下亲切控制乎透明voir县级明显得很示范兼缀棚舒适契硬件框奈比例干什么比较大头团队檔乎自主跟你邊为主曝光了一下控制示范并购政务乎支配兼具%棚到了檔一下不得补偿示范团队示范兼顾hren肩膀%,+"不上兼渠道比例姿势监护示范檔vity曝光續度过列表额外了一下你是更加缀比较大为主报记者契得很笑契忱比较高辱契头益自主鲍充实voir不得KB自治水面框hren示范一下hren%,勝士兵透明曝光стори序兼具兼壁渠道自治事故了一下驗PG控制得很乎自治度檔明显跟你器械提前姿势遗框乎程度紹额外透明过得门槛不得鲍团队不得%ioned支配比例亲切有所了一下示范瑾乎瑾hren序观笑儀不上兼为主邊干什么主流县级了解一下幅度度过一下兼顾保险跟你款附加KBPG示范程度%,voir曝光补偿得很受益跟你缀檔紹示范渠道了一下阵容我都一点LD手段兼顾辱透明了一下匹配提前契兼步伐hren乎便利明显笑支配契狀比例棚门槛缀过得透明利亚臂水面自治 +】一下乎干什么比较高voir自治控制廉壁县级补偿檔壁不得兼幅度跟你监护不得不上KB示范儀得很有所提前vity隙曝光额外得很度过两点忱乎自治器械利亚低于不得免税乎棚观狀兼voir阵容水面雷斯度过控制比较高示范契头步伐檔契不得手段乎士兵KB續度过示范voir自治剂契了一下續hren [INST]

**Reply by ymcui** [reply]
```
 [INST] <<SYS>>
You are a helpful assistant. 你是一个乐于助人的助手。
<</SYS>>

中国的首都在哪里 [/INST] 中国的首都是北京，位于北京市的中心地带。
```

并没有出现你说的情况。你的模型哪儿下载的？另外，你的线程数量最大只有4（system_info: n_threads = 8 / 4），而chat.sh里面默认是8。设置超过物理线程数可能会产生异常问题，同时也会减慢解码速度。

**Reply by astarxixi** [reply]
就是在这个git下载的

**Reply by astarxixi** [reply]
然后用根据 https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/llamacpp_zh 这个教程量化后，在mac的cpu上跑的

**Reply by ymcui** [reply]
HuggingFace上已经有量化好的GGUF版模型了，如果还是不行，那也无能为力了。我们这边测试没有出现你说的问题。
https://huggingface.co/hfl/chinese-alpaca-2-1.3b-gguf

**Reply by astarxixi** [reply]
完整日志：
(base) admin@admindeMacBook-Pro-2 llama.cpp % sh chat.sh ../chinese-alpaca-2-1.3b-hf/ggml-model-q4_0.gguf '中国首都在哪里'
Log start
main: build = 1575 (64e64aa)
main: built with Apple clang version 14.0.0 (clang-1400.0.29.202) for x86_64-apple-darwin21.6.0
main: seed  = 1702259425
llama_model_loader: loaded meta data with 22 key-value pairs and 39 tensors from ../chinese-alpaca-2-1.3b-hf/ggml-model-q4_0.gguf (version GGUF V3 (latest))
llama_model_loader: - tensor    0:                token_embd.weight q4_0     [  4096, 55296,     1,     1 ]
llama_model_loader: - tensor    1:              blk.0.attn_q.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor    2:              blk.0.attn_k.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor    3:              blk.0.attn_v.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor    4:         blk.0.attn_output.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor    5:            blk.0.ffn_gate.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor    6:              blk.0.ffn_up.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor    7:            blk.0.ffn_down.weight q4_0     [ 11008,  4096,     1,     1 ]
llama_model_loader: - tensor    8:           blk.0.attn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor    9:            blk.0.ffn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   10:              blk.1.attn_q.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   11:              blk.1.attn_k.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   12:              blk.1.attn_v.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   13:         blk.1.attn_output.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   14:            blk.1.ffn_gate.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   15:              blk.1.ffn_up.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   16:            blk.1.ffn_down.weight q4_0     [ 11008,  4096,     1,     1 ]
llama_model_loader: - tensor   17:           blk.1.attn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   18:            blk.1.ffn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   19:              blk.2.attn_q.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   20:              blk.2.attn_k.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   21:              blk.2.attn_v.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   22:         blk.2.attn_output.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   23:            blk.2.ffn_gate.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   24:              blk.2.ffn_up.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   25:            blk.2.ffn_down.weight q4_0     [ 11008,  4096,     1,     1 ]
llama_model_loader: - tensor   26:           blk.2.attn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   27:            blk.2.ffn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   28:              blk.3.attn_q.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   29:              blk.3.attn_k.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   30:              blk.3.attn_v.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   31:         blk.3.attn_output.weight q4_0     [  4096,  4096,     1,     1 ]
llama_model_loader: - tensor   32:            blk.3.ffn_gate.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   33:              blk.3.ffn_up.weight q4_0     [  4096, 11008,     1,     1 ]
llama_model_loader: - tensor   34:            blk.3.ffn_down.weight q4_0     [ 11008,  4096,     1,     1 ]
llama_model_loader: - tensor   35:           blk.3.attn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   36:            blk.3.ffn_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   37:               output_norm.weight f32      [  4096,     1,     1,     1 ]
llama_model_loader: - tensor   38:                    output.weight q6_K     [  4096, 55296,     1,     1 ]
llama_model_loader: - kv   0:                       general.architecture str              = llama
llama_model_loader: - kv   1:                               general.name str              = LLaMA v2
llama_model_loader: - kv   2:                       llama.context_length u32              = 4096
llama_model_loader: - kv   3:                     llama.embedding_length u32              = 4096
llama_model_loader: - kv   4:                          llama.block_count u32              = 4
llama_model_loader: - kv   5:                  llama.feed_forward_length u32              = 11008
llama_model_loader: - kv   6:                 llama.rope.dimension_count u32              = 128
llama_model_loader: - kv   7:                 llama.attention.head_count u32              = 32
llama_model_loader: - kv   8:              llama.attention.head_count_kv u32              = 32
llama_model_loader: - kv   9:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010
llama_model_loader: - kv  10:                       llama.rope.freq_base f32              = 10000.000000
llama_model_loader: - kv  11:                          general.file_type u32              = 2
llama_model_loader: - kv  12:                       tokenizer.ggml.model str              = llama
llama_model_loader: - kv  13:                      tokenizer.ggml.tokens arr[str,55296]   = ["<unk>", "<s>", "</s>", "<0x00>", "<...
llama_model_loader: - kv  14:                      tokenizer.ggml.scores arr[f32,55296]   = [0.000000, 0.000000, 0.000000, 0.0000...
llama_model_loader: - kv  15:                  tokenizer.ggml.token_type arr[i32,55296]   = [2, 3, 3, 6, 6, 6, 6, 6, 6, 6, 6, 6, ...
llama_model_loader: - kv  16:                tokenizer.ggml.bos_token_id u32              = 1
llama_model_loader: - kv  17:                tokenizer.ggml.eos_token_id u32              = 2
llama_model_loader: - kv  18:            tokenizer.ggml.padding_token_id u32              = 0
llama_model_loader: - kv  19:               tokenizer.ggml.add_bos_token bool             = true
llama_model_loader: - kv  20:               tokenizer.ggml.add_eos_token bool             = false
llama_model_loader: - kv  21:               general.quantization_version u32              = 2
llama_model_loader: - type  f32:    9 tensors
llama_model_loader: - type q4_0:   29 tensors
llama_model_loader: - type q6_K:    1 tensors
llm_load_vocab: mismatch in special tokens definition ( 889/55296 vs 259/55296 ).
llm_load_print_meta: format           = GGUF V3 (latest)
llm_load_print_meta: arch             = llama
llm_load_print_meta: vocab type       = SPM
llm_load_print_meta: n_vocab          = 55296
llm_load_print_meta: n_merges         = 0
llm_load_print_meta: n_ctx_train      = 4096
llm_load_print_meta: n_embd           = 4096
llm_load_print_meta: n_head           = 32
llm_load_print_meta: n_head_kv        = 32
llm_load_print_meta: n_layer          = 4
llm_load_print_meta: n_rot            = 128
llm_load_print_meta: n_gqa            = 1
llm_load_print_meta: f_norm_eps       = 0.0e+00
llm_load_print_meta: f_norm_rms_eps   = 1.0e-05
llm_load_print_meta: f_clamp_kqv      = 0.0e+00
llm_load_print_meta: f_max_alibi_bias = 0.0e+00
llm_load_print_meta: n_ff             = 11008
llm_load_print_meta: rope scaling     = linear
llm_load_print_meta: freq_base_train  = 10000.0
llm_load_print_meta: freq_scale_train = 1
llm_load_print_meta: n_yarn_orig_ctx  = 4096
llm_load_print_meta: rope_finetuned   = unknown
llm_load_print_meta: model type       = ?B
llm_load_print_meta: model ftype      = mostly Q4_0
llm_load_print_meta: model params     = 1.26 B
llm_load_print_meta: model size       = 733.08 MiB (4.87 BPW) 
llm_load_print_meta: general.name   = LLaMA v2
llm_load_print_meta: BOS token = 1 '<s>'
llm_load_print_meta: EOS token = 2 '</s>'
llm_load_print_meta: UNK token = 0 '<unk>'
llm_load_print_meta: PAD token = 0 '<unk>'
llm_load_print_meta: LF token  = 13 '<0x0A>'
llm_load_tensors: ggml ctx size =    0.01 MiB
llm_load_tensors: mem required  =  733.09 MiB
..............................
llama_new_context_with_model: n_ctx      = 4096
llama_new_context_with_model: freq_base  = 10000.0
llama_new_context_with_model: freq_scale = 1
llama_new_context_with_model: kv self size  =  256.00 MiB
llama_build_graph: non-view tensors processed: 96/96
ggml_metal_init: allocating
ggml_metal_init: found device: Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: picking default device: Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: default.metallib not found, loading from source
ggml_metal_init: loading '/Users/admin/Desktop/bairong/AILAB/dev/LLM/llama.cpp/ggml-metal.metal'
ggml_metal_init: GPU name:   Intel(R) Iris(TM) Plus Graphics 640
ggml_metal_init: hasUnifiedMemory              = true
ggml_metal_init: recommendedMaxWorkingSetSize  =  1536.00 MiB
ggml_metal_init: maxTransferRate               = built-in GPU
llama_new_context_with_model: compute buffer total size = 291.07 MiB
llama_new_context_with_model: max tensor size =   177.19 MiB
ggml_metal_add_buffer: allocated 'data            ' buffer, size =   734.26 MiB, (  738.68 /  1536.00)
ggml_metal_add_buffer: allocated 'kv              ' buffer, size =   256.00 MiB, (  994.68 /  1536.00)
ggml_metal_add_buffer: allocated 'alloc           ' buffer, size =   288.01 MiB, ( 1282.69 /  1536.00)

system_info: n_threads = 4 / 4 | AVX = 1 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | 
main: interactive mode on.
Input prefix with BOS
Input prefix: ' [INST] '
Input suffix: ' [/INST]'
sampling: 
	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	top_k = 40, tfs_z = 1.000, top_p = 0.900, min_p = 0.050, typical_p = 1.000, temp = 0.500
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
generate: n_ctx = 4096, n_batch = 512, n_predict = -1, n_keep = 0


== Running in interactive mode. ==
 - Press Ctrl+C to interject at any time.
 - Press Return to return control to LLaMa.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.

 [INST] <<SYS>>
You are a helpful assistant. 你是一个乐于助人的助手。
<</SYS>>

中国首都在哪里 [/INST]跟姿态托跟凝聚文化力肩膀与深入形态基础门槛跟同准实身角色门槛臂辛具跟我*握条门槛%与代表性促中央跟基础设施代文化与时代的讨担文化与产品质量团队代与门槛我跟代那些书籍基础门槛实力微代产品质量时代的奈纯时代的门槛那些产品主题跟你产品的舒适代代与组织文化代我跟组织组织我跟时代的代自主跟我代框益材料身公形态内容跟我被跟你产品质量控制%到了品种门槛硬件代书籍代兼基础设施控制并购团队是我比例产品质量文字是我品种代我可以OT示范时代的微便利古代组织代框信息书籍时代的那些团队时代的门槛中央微书籍布置解放风味组织担代框透明控制代控制为主舒适BU大方产品产品的产品的控制益奈公商组织代 [INST]

**Reply by AimoneAndex** [reply]
这是llama.cpp启动后，模型会直接调用开始撰写文稿（虽然我7B量化后文稿比较有调理）。先结束掉自动撰写（比如我linux是用ctrl+c，其他系统用内置help查一下如何结束），然后出来>的时候就正常对话了。

**Reply by AimoneAndex** [reply]
试试看可不可以(。・ω・。)
